shared/tree-sitter-extractor/src/extractor/simple.rs

```diff
@@ -83,16 +82,23 @@ impl Extractor {
             schemas.push(schema);
         }
 
-        // Construct a map from file extension -> LanguageSpec
-        let mut file_extension_language_mapping: HashMap<&OsStr, Vec<usize>> = HashMap::new();
-        for (i, lang) in self.languages.iter().enumerate() {
-            for (j, _ext) in lang.file_extensions.iter().enumerate() {
-                let indexes = file_extension_language_mapping
-                    .entry(&lang.file_extensions[j])
-                    .or_default();
-                indexes.push(i);
+        // Construct a single globset containing all language globs,
+        // and a mapping from glob index to language index.
+        let (globset, glob_language_mapping) = {
+            let mut builder = GlobSetBuilder::new();
```

Comment:
Let's set the literal_separator feature, so path separators do not match the `*` and `?` wildcards.
```suggestion
            let mut builder = GlobSetBuilder::new().literal_separator(true);
```

Reply:
Sure, but if you don't want that then you should match the globs against the whole paths. Instead of `globset.matches(&path);` we could do something like `globset.matches(&path.filename());